lua/lang/cpp.lua

```diff
@@ -1,24 +1,22 @@
 local M = {}
 
 M.config = function()
-  O.lang.clang = {
+  O.lang.cpp = {
     diagnostics = {
       virtual_text = { spacing = 0, prefix = "" },
       signs = true,
       underline = true,
     },
-    cross_file_rename = true,
-    header_insertion = "never",
     filetypes = { "c", "cpp", "objc" },
-    formatter = {
-      exe = "clang-format",
-      args = {},
-      stdin = true,
-    },
-    linters = {
-      "cppcheck",
-      "clangtidy",
+    clangd_flags = {
+      "--background-index",
+      "--header-insertion=never",
+      "--cross-file-rename",
+      "--clang-tidy",
+      "--clang-tidy-checks=-*,llvm-*,clang-analyzer-*,modernize-*",
     },
+    formatter = {},
+    linters = {},
```

Comment:
these lines pretty much make `format_on_save` and `lint_on_save` useless, do we want that?
or do we want to leave it as is and fix #996 ?

Reply:
IMO we should let the user have the choice.
Also, clangd doesn't run  cppcheck.